### ISSUE TITLE
Waiting time should be limited

### ISSUE THREAD

**Issue by blokhin** [tilde-lab/metis-client#11]
Currently, in case of some errors, this waits forever, and the synchronous client hangs on:

```
  ...
  File "/home/eb/metis-client/lib/python3.8/site-packages/asgiref/sync.py", line 218, in __call__
    current_executor.run_until_future(loop_future)
  File "/home/eb/metis-client/lib/python3.8/site-packages/asgiref/current_thread_executor.py", line 59, in run_until_future
    work_item = self._work_queue.get()
  File "/usr/lib/python3.8/queue.py", line 170, in get
    self.not_empty.wait()
  File "/usr/lib/python3.8/threading.py", line 302, in wait
    waiter.acquire()
```

Probably, some timer should be presented.

**Comment by blokhin** [comment]
to reproduce, e.g. stop BFF during the test run of client... it waits forever...

**Comment by blokhin** [comment]
Wow, I have also discovered it might influence the GUI for that particular user :flushed:

![metis-gui-bug](https://user-images.githubusercontent.com/177388/236035119-504a53f5-986d-4332-91fe-0c1ba5912804.png)

this happens only when our script hangs... As soon as we stop the script, GUI starts to work normally. Obviously, BFF works.

**Comment by blokhin** [comment]
@knopki please, let’s prioritise this 🙏

**Comment by knopki** [comment]
> Wow, I have also discovered it might influence the GUI for that particular user flushed
> this happens only when our script hangs... As soon as we stop the script, GUI starts to work normally. Obviously, BFF works.

I can't confirm that.
If BFF dies because of some error, then yes, `metis-client` might influence the GUI.
But if the synchronous client is just waiting for results, the GUI works just fine.
Therefore, if there is a bug here, it's on the BFF side.

Timeouts as a client feature are implemented in #23
